### ISSUE TITLE
[Bug] The tabs tray FAB will appear in Private mode when accessibility mode is enabled

### ISSUE THREAD

**Issue by LaurentiuApahideanSV** [mozilla-mobile/fenix#19938]
## Steps to reproduce
1. Turn on accessibility mode.
2. Open the tabs tray.
3. Switch from normal browsing to private browsing
### Expected behavior
The tabs tray FAB should not appear.
### Actual behavior
The tabs tray FAB appear and has no function.
### Device information

* Device vendor / model and Android version: OnePlus 6T (Android 9), Google Pixel 3 XL (Android 9)
* Firefox for Android version: Firefox Preview Nightly 6/10
![FAB_Accessibility](https://user-images.githubusercontent.com/50983519/121526298-e9ba2280-ca01-11eb-864d-04b6debc1707.gif)

**Comment by mcarare** [comment]
@jonalmeida 
On testing this, I noticed that if I remove the accessibility check the Tab tray would work as expected, meaning Talkback can select the FAB. The way it was implemented before made the FAB unreachable on a11y tree scanning, but the refactoring seems to have fixed this.

Given this, do we really need the `AccessibleNewTabButtonBinding` and button? 

Note: The issue in this bug is that the `onState` is never called because `super.start()` is not called when a11y services are enabled.

**Comment by jonalmeida** [comment]
> On testing this, I noticed that if I remove the accessibility check the Tab tray would work as expected, meaning Talkback can select the FAB. The way it was implemented before made the FAB unreachable on a11y tree scanning, but the refactoring seems to have fixed this.
> 
> Given this, do we really need the `AccessibleNewTabButtonBinding` and button?

This is another great win for refactoring the tabs tray! ✨  

Yeah, let's kill it in that case and document why removed it as well (on the PR comment of this issue I guess?) so that we know why.

**Comment by mcarare** [comment]
Note for QA: The tabs tray should be the same with or without a11y enabled. The testing should ensure the FAB works as expected.

**Comment by lobontiumira** [comment]
Verified as fixed on the 6/16 Nightly build with Google Pixel (Android 10).
The tabs tray is displayed the same, with or without talkback/voice assistant enabled.
